core/import-seac.c

```diff
@@ -0,0 +1,360 @@
+	struct tm divetime = {0};
+	time_t tmdivetime;
+	struct gasmix lastgas, curgas;
+	struct parser_state *state = (struct parser_state *)param;
+	sqlite3 *handle = state->sql_handle;
+	sqlite3_stmt *sqlstmt;
+
+	const char *get_samples = "SELECT dive_number,runtime_s,depth_cm, temperature_mCx10,active_O2_fr, first_stop_depth_cm, first_stop_time_s, ndl_tts_s,cns FROM dive_data WHERE dive_number = ? ORDER BY runtime_s ASC";
+		//  0 = dive_number
+		//  1 = runtime_s
+		//  2 = depth_cm
+		//  3 = temperature_mCx10  - eg dC
+		//  4 = active_O2_fr
+		//  5 = first_stop_depth_cm
+		//  6 = first_stop_time_s
+		//  7 = ndl_tts_s
+		//  8 = cns
+
+	dive_start(state);
+	state->cur_dive->number = atoi(data[0]);
+
+	// Get time and date
+	sscanf(data[2],"%d/%d/%2d",&day,&month,&year);
+	sscanf(data[4],"%2d:%2d:%2d",&hour,&mins,&secs);
+
+	// TODO: Confirm Seac behaviour timezones
+	// Suspect a lookup table is used.
+	tz = atoi(data[3]);
+
+	// Timezone offset lookup array
+	const int timezoneoffset[][2] =
+	{{-12,   0},
+	 {-11,   0},
+	 {-10,   0},
+	 {-9 , -30},
+	 {-9 ,   0},
+	 {-8 ,   0},
+	 {-7 ,   0},
+	 {-6 ,   0},
+	 {-5 ,   0},
+	 {-4 , -30},
+	 {-4 ,   0},
+	 {-3 , -30},
+	 {-3 ,   0},
+	 {-2 ,   0},
+	 {-1 ,   0},
+	 { 0 ,   0},
+	 { 1 ,   0},
+	 { 2 ,   0},
+	 { 3 ,   0},
+	 { 3 ,  30},
+	 { 4 ,   0},
+	 { 4 ,  30},
+	 { 5 ,   0},
+	 { 5 ,  30},
+	 { 5 ,  45},
+	 { 6 ,   0},
+	 { 6 ,  30},
+	 { 7 ,   0},
+	 { 8 ,   0},
+	 { 8 ,  45},
+	 { 9 ,   0},
+	 { 9 ,  30},
+	 { 9 ,  45},
+	 { 10,   0},
+	 { 10,  30},
+	 { 11,   0},
+	 { 11,  30},
+	 { 12,   0},
+	 { 12,  45},
+	 { 13,   0},
+	 { 13,  45},
+	 { 14,   0}};
+
+	hour = hour + timezoneoffset[tz][0];
+	mins = mins + timezoneoffset[tz][1];
+
+	// Deal with minute over/underflow
+	if (mins > 59){
+		mins -= 60;
+		hour += 1;
+	} else if (mins < 0){
+		mins += 60;
+		hour -= 1;
+	}
+
+	// Handle hour overflow
+	if (hour < 0){
+		hour += 24;
+		day -= 1;
+	} else if (hour > 23){
+		hour -= 24;
+		day += 1;
+	}
+
+	// Year is scanned in as two digits, so add 2000
+	year += 2000;
+	// Leap year check
+	if (!(year % 400)){
+	    isleap = true;
+	} else if (!(year % 100)){
+		isleap = false;
+	} else if (!(year % 4)){
+		isleap = true;
+	}
+
+	// Handle Day overflow
+	if (day > 31){
+		day -= 31;
+		month += 1;
+	// remember, months are as scanned in here
+	} else if (day > 30 && (month == 4 || month == 6 || month == 9 || month == 11)){
+		day -= 30;
+		month += 1;
+	} else if (day > 28 && month == 2 && isleap == false){
+		day -= 28;
+		month += 1;
+	} else if (day > 29 && month == 2 && isleap == true){
+		day -= 29;
+		month += 1;
+	}
+
+
+	// Handle month overflow
+	if (month > 12){
+		year += 1;
+		month -= 12;
+	}
+
+	divetime.tm_year = year - 1900;
+	divetime.tm_mon = month - 1;
+	divetime.tm_mday = day;
+	divetime.tm_hour = hour;
+	divetime.tm_min = mins;
+	// WARNING: Possibly Non portable
+	divetime.tm_sec = secs - timezone; // counteract mktime() assumption of local time
```

Comment:
I am not really familiar with this timezone variable. However, it seems to be defined as seconds west of UTC (man tzset). If this is the case, UTC+1 would mean timezone == -3600. Your code suggests that I have misunderstood the whole thing. So explanation would be in order. Or fixing the code :D 

Of course, it must also be ensured that this works on Windows also.

Reply:
I added a large comment block in the next commit.

timezone comes from <time.h>, and you're correct, it's seconds west of UTC.
Unfortunately mktime() assumes it is passed local time in the tm struct, so it applies the offset
of the local machine when returning time_t . Have to compensate for this as the timezone
for the dive doesn't necessarily match timezone of user's machine.

Reply:
Just a quick remark: In other places (the image importer), we use Qt's functions for date/time calculations with time zones etc, so if this doesn't work on windows, you could try that as well. The problem is, you cannot directly call Qt functionality from pure C (as opposed to C++). So when we need to do that, we go through a helper function in qthelper.cpp.

Reply:
I've looked into mktime() further, and it seems there are even more pitfalls involving if the local machine is currently in DST. The times end up being 1hr off.

I think I'm going to have to dig into the QT DateTime library a little bit.